tests/unit/test-class-wc-payment-gateway-wcpay.php

```diff
@@ -812,6 +814,24 @@ public function test_capture_charge_success() {
 		$order->update_meta_data( '_intention_status', 'requires_capture' );
 		$order->update_status( 'on-hold' );
 
+		$mock_intent = new WC_Payments_API_Intention(
```

Comment:
most of the test methods looks the same. I wonder if we can use a dataProivder. I know that the test methods were in place, but if we can use a data provider we can do a refactor.

Reply:
True. Thanks for pointing it out. I'll see if I could do the refactor.